internal/nimparser/nimparser.go

```diff
@@ -152,11 +157,17 @@ func MatchProfiles(modelSpec appsv1alpha1.ModelSpec, manifest NIMManifest, disco
 			continue
 		}
 
-		// GPU matching logic
-		if len(modelSpec.GPUs) > 0 || len(discoveredGPUs) > 0 {
-			if !matchGPUProfile(modelSpec, profile, discoveredGPUs) {
+		// Perform GPU match only when optimized engine is selected or GPU filters are provided
+		if isOptimizedEngine(modelSpec.Engine) || len(modelSpec.GPUs) > 0 {
```

Comment:
A model may not use tensorrt engine while still requires gpu, is that right? Seems that for that case there would be no match

Reply:
No, this case is to still filter optimized profiles, when GPU options are provided, but not the engine type itself in the model spec.